clang/include/clang/CIR/Dialect/IR/CIRDialect.td

```diff
@@ -0,0 +1,44 @@
+//===- CIRDialect.td - CIR dialect -------------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the CIR dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_CIR_DIALECT_IR_CIRDIALECT
+#define LLVM_CLANG_CIR_DIALECT_IR_CIRDIALECT
+
+include "mlir/IR/OpBase.td"
+
+def CIR_Dialect : Dialect {
+  let name = "cir";
+
+  // A short one-line summary of our dialect.
+  let summary = "A high-level dialect for analyzing and optimizing Clang "
+                "supported languages";
+
+  let cppNamespace = "::mlir::cir";
+
+  let useDefaultAttributePrinterParser = 0;
+  let useDefaultTypePrinterParser = 0;
+
+  let extraClassDeclaration = [{
+    void registerAttributes();
+    void registerTypes();
+
+    Type parseType(DialectAsmParser &parser) const override;
```

Comment:
Because this is new code, the parameter names should match the usual coding style for naming (start with a capital letter) for consistency with the rest of the code base.

Where do the definitions for these come from?

Reply:
This will be added to generated class which have lower-case starting APIs: consistency within the class seems to prevail to me.
You should really expect that all the MLIR code will follow this convention, because it'll be intertwined closely with the MLIR framework.

Reply:
> This will be added to generated class which have lower-case starting APIs: consistency within the class seems to prevail to me.

Ah, then yeah, I agree. Thank you!

> You should really expect that all the MLIR code will follow this convention, because it'll be intertwined closely with the MLIR framework.

MLIR isn't old enough to really get "grandfathered" in with their own style, so that's pretty surprising. That said, I *fully support* letting projects have project-specific policies including coding style, so I see no reason for MLIR to change either. Interfacing closely with MLIR does make this a bit awkward however, because it's also intertwined with the Clang code base. On balance, I think it's fine to follow the MLIR convention because I think there will be more surface area in common between CIR + MLIR than CIR + Clang (and Clang has plenty of inconsistency with coding style already so this is not really novel).

tl;dr: nothing needs to change here.

Reply:
I think I mentioned it elsewhere, but worth keeping in mind: MLIR didn't intend to diverge from LLVM, it started about at the time where there was this plan to adopt a new style for LLVM: https://llvm.org/docs/Proposals/VariableNames.html
So we thought we were just adopting the "new coding style" to limit churn, and unfortunately this didn't pan out for the whole codebase (I think lldb and lld are also using this style now)